analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Monitor.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Threading;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        private object obj = new object();
+
+        public void DoSomethingWithMonitor1(bool b)
+        {
+            Monitor.Enter(obj); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor2(bool b)
+        {
+            Monitor.Enter(obj); // FN
+            switch (b)
+            {
+                case true:
+                    Monitor.Exit(obj);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public void DoSomethingWithMonitor3()
+        {
+            Monitor.Enter(obj); // FN
+            var a = new Action(() =>
+            {
+                Monitor.Exit(obj);
+            });
+        }
+
+        public int MyProperty
+        {
+            set
+            {
+                Monitor.Enter(obj); // FN
+                if (value == 42)
+                {
+                    Monitor.Exit(obj);
+                }
+            }
+        }
+
+        public void DoSomethingWithMonitor4()
+        {
+            Monitor.Enter(obj); // FN
+            for (int i = 0; i < 10; i++)
+            {
+                if (i == 9)
+                {
+                    Monitor.Exit(obj);
+                }
+            }
+        }
+
+        public void DoSomethingWithMonitor5(string b)
```

Comment:
`string` and `b`? Looks weird after `bool b`. We can make them all `arg` if they don't have a special meaning.
```suggestion
        public void DoSomethingWithMonitor5(string arg)
```